src/borg/remote.py

```diff
@@ -345,8 +345,11 @@ def open(self, path, create=False, lock_wait=None, lock=True, exclusive=None, ap
                     break
             else:
                 raise PathNotAllowed(path)
+        # borg init on borg serve --append-only does not create an append only repo,
+        # while borg init --append-only does, regardless of the --append-only flag for serve.
+        append_only = (not create and self.append_only) or append_only
```

Comment:
I suspect you have analyzed the precise meaning of self.append_only vs. append_only to make sure this condition is correct, so maybe add some docs to the respective places about what they mean precisely (how they get True).